.github/workflows/kickstart-tests.yml

```diff
@@ -0,0 +1,186 @@
+          git rebase origin/${{ env.TARGET_BRANCH }}
+
+      - name: Clone tclib repository
+        uses: actions/checkout@v3
+        with:
+          repository: rhinstaller/tclib
+          path: tclib
+
+      - name: Clone kickstart-tests repository
+        uses: actions/checkout@v3
+        with:
+          repository: rhinstaller/kickstart-tests
+          path: kickstart-tests
+
+      - name: Generate tclib test cases
+        working-directory: ./kickstart-tests
+        run: scripts/generate-testcases.py -t ./testlib/test_cases/kstest-template.tc.yaml.j2 . -o ./testlib/test_cases
+
+      - name: Ensure http proxy is running
+        run: sudo kickstart-tests/containers/squid.sh start
+
+      - name: Generate query arguments
+        id: generate_query
+        working-directory: ./kickstart-tests
+        run: |
+          set -eux
+          PERMIAN_QUERY=$(scripts/generate-permian-query.py ${TEST_SELECTION})
+          if [ $? == 0 ]; then
+            echo "::set-output name=query::$PERMIAN_QUERY"
+          else
+            echo "Parsing of the request arguments failed"
+            exit 1
+          fi
+
+      - name: Configure Permian event values
+        id: set_event_values
+        run: |
+          set -eux
+          echo "::set-output name=installation_tree::${REPOSITORY}/Everything/x86_64/os/"
+          echo "::set-output name=modular_url::${REPOSITORY}/Modular/x86_64/os/"
+          echo "::set-output name=platform::${PLATFORM}"
+
+      - name: Run kickstart tests in container
+        working-directory: ./permian
+        run: |
+          sudo --preserve-env=TEST_JOBS \
+          PYTHONPATH=${PYTHONPATH:-}:${{ github.workspace }}/tclib \
+          ./run_subset --debug-log permian.log \
+            -o library.directPath="${{ github.workspace }}/kickstart-tests/testlib" \
+            -o workflows.dry_run=False \
+            -o kickstart_test.kstest_local_repo="${{ github.workspace }}/kickstart-tests" \
+            --testcase-query '${{ steps.generate_query.outputs.query }}' \
+            run_event '{"type":"github.pr","kstestParams":{"platform":"${{ steps.set_event_values.outputs.platform }}","urls":{"x86_64":{"installation_tree":"${{ steps.set_event_values.outputs.installation_tree }}","modular_url":"${{ steps.set_event_values.outputs.modular_url }}"}}}}'
+
+      - name: Collect anaconda logs
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: 'logs'
+          # skip the /anaconda subdirectories, too large
+          path: |
+            kickstart-tests/data/logs/kstest.log
+            kickstart-tests/data/logs/kstest-*/*.log
+            kickstart-tests/data/additional_repo/*.rpm
+
+      - name: Collect Permian logs
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: 'logs-permian'
+          path: |
+            permian/permian.log
+
+      # Permian hides the exit code of launcher
```

Comment:
We need to implement GitHub reportSender ASAP.

Reply:
Thank you for that! I've updated the PR to use it.